packages/theme-patternfly-org/scripts/md/mdx-hast-to-jsx.js

```diff
@@ -0,0 +1,185 @@
+  if (node.type === 'text') {
+    return serializeText(node, options, parentNode);
+  }
+
+  if (node.type === 'mdxBlockExpression' || node.type === 'mdxSpanExpression') {
+    return serializeMdxExpression(node);
+  }
+
+  if (node.type === 'mdxBlockElement' || node.type === 'mdxSpanElement') {
+    return serializeComponent(node, options, parentNode);
+  }
+
+  if (node.type === 'import') {
+    return serializeEsSyntax(node);
+  }
+}
+
+function compile(options = {}) {
+  this.Compiler = function (tree) {
+    return toJSX(tree, {}, options);
+  }
+}
+
+function serializeRoot(node, options) {
+  const { getRelPath, getPageData } = options;
+  const pageData = {...getPageData()};
+  const exportName = pageData.slug.replace(/[\/-](.)?/g, (_, match) => capitalize(match)) + 'Docs';
+  options.exportName = exportName;
+  const groups = {
+    import: [],
+    rest: []
+  };
+
+  for (const child of node.children) {
+    let kind = 'rest'
+
+    if (child.type === 'import') {
+      kind = child.type
+    }
+
+    groups[kind].push(child)
+  }
+
+  const importStatements = groups.import
+    .map(childNode => toJSX(childNode, node))
+    .map(imp => imp.replace(/(['"])\./g, (_, match) => `${match}${getRelPath()}`))
+    .join('\n')
+
+  // https://astexplorer.net/#/gist/9c531dd372dfc57e194c13c2889d31c3/03f2d6e889db1a733c6a079554e8af7784863739
+  const importSpecifiers = jsxParser.parse(importStatements, { sourceType: 'module' }).body
+    .map(node => node.specifiers)
+    .flat(1)
+    .map(spec => spec.local ? spec.local.name : null)
+    .filter(localName => !/srcImport\d+/.test(localName)) // Images in MD like [!img](./src)
+    .filter(Boolean)
+    .join(',\n  ');
+
+  const childNodes = groups.rest
+    .map(childNode => toJSX(childNode, node, options))
+    .join('');
+
+  return `import React from 'react';
+import { AutoLinkHeader, Example, Link as PatternflyThemeLink } from 'theme-patternfly-org/components';
+${importStatements}
+
+export const ${exportName} = ${JSON.stringify(pageData, null, 2)};
+${exportName}.liveContext = {
+  ${importSpecifiers}
+};
+${exportName}.Component = () => (
+  <React.Fragment>${childNodes.replace(/\n\s*\n/g, '\n')}
+  </React.Fragment>
+);
+`;
+}
+
+function serializeElement(node, options) {
+  const { indent, exportName } = options;
+  const {type, props} = toH(
+    fakeReactCreateElement,
+    Object.assign({}, node, {children: []}),
+    {prefix: false}
+  );
+  const content = serializeChildren(node, {...options});
+
+  delete props.key;
+
+  const srcImport = node.properties.src;
+  if (type === 'img') {
+    delete props.src;
+  }
+  const spread = Object.keys(props).length === 0
+    ? null
+    : JSON.stringify(props);
+
+  const indentText = '  '.repeat(indent);
+
+  return `
+${indentText}<${type}${
+  type === 'Example' ? ` {...${exportName}}` : ''}${
+  type === 'img' && srcImport ? ` src={${srcImport}}` : ''}${
+  spread ? ` {...${spread}}` : ''}>
+${indentText}  ${content}
```

Comment:
```suggestion
${indentText}${indentText}${content}
```

Reply:
It's actually significant that on that line one extra indent is added.

Reply:
yes, but depending on the intent option it could be more than 2 spaces right? which is why I suggested repeating `indentText` twice

Reply:
Oh, well I assume you always want a single indent (which I've defined as two spaces) for JSX children. So at `indent=3` you'll see:
```jsx
      <Tag>
        {children}
      <Tag>
```

If I were to accept your change it'd be:
```jsx
      <Tag>
            {children}
      <Tag>
```

Reply:
that's how it works typically, if you set VSCode intendation to 4, every indentation gives you 4 spaces, anyways just something that stood out to me, not sure if it matters for our docs

Reply:
Oh, sorry the variable name `indent` must be confusing you. It really is the indentation _level_, not the number of spaces to add. So at `indent=3` there are `'  '.repeat(3)` = '      '  = 6 spaces.

Sorry, that's confusing! It's really just a best attempt to format the generated docs. Something proper would be to run the code through babel/prettier but that's rather expensive for files that not many people are looking at anyway and I figured this could suffice for now.